akka-stream/src/main/scala/akka/stream/scaladsl/Compression.scala

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.scaladsl
+
+import akka.NotUsed
+import akka.stream.impl.io.compression._
+import akka.util.ByteString
+
+object Compression {
+  /**
+   * Creates a flow that gzip-compresses a stream of ByteString. Note that the compressor
+   * will SYNC_FLUSH'd after every [[ByteString]] so that it is guaranteed that every [[ByteString]]
+   * coming out of the flow can be fully decompressed without waiting for additional data. This may
+   * come at a compression performance cost for very small chunks.
+   *
+   * FIXME: should compression level / strategy / flush mode be configurable?
+   */
+  def gzip: Flow[ByteString, ByteString, NotUsed] =
```

Comment:
yes would be good, please add an overloaded method that takes a GzipCompressionCOnfig object.
Overload instead of default param because easier to binary compatibility evolve afterwards